common/gcp.go

```diff
@@ -3,12 +3,66 @@ package common
 import (
 	"context"
 	"fmt"
+	"io/ioutil"
 	"net/http"
+	"os"
 
 	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/idtoken"
 	"google.golang.org/api/impersonate"
+	"google.golang.org/api/option"
 )
 
+// Configures an authorizer that uses credentials sourced from JSON or file.
+// This auth mode does NOT use impersonation and it does NOT use Application
+// Default Credentials (ADC).
+func (c *DatabricksClient) configureWithGoogleCrendentials(
+	ctx context.Context) (func(r *http.Request) error, error) {
+	if c.GoogleServiceAccount == "" || c.GoogleCredentials == "" || !c.IsGcp() {
+		return nil, nil
+	}
+	if c.Host == "" {
+		return nil, fmt.Errorf("'host' field is empty or not set. "+
+			"Provider config <%s>", c.configDebugString())
+	}
+	json, err := readCredentials(c.GoogleCredentials)
+	if err != nil {
+		err = fmt.Errorf("could not read GoogleCredentials. "+
+			"Make sure the file exists, or the JSON content is valid: %w", err)
+		return nil, err
+	}
+	// Obtain token source for creating OIDC token.
+	audience := c.Host
+	oidcSource, err := idtoken.NewTokenSource(ctx, audience, option.WithCredentialsJSON([]byte(json)))
+	if err != nil {
+		return nil, fmt.Errorf("could not obtain OIDC token from JSON: %w", err)
+	}
+	// Obtain token source for creating Google Cloud Platform token.
+	creds, err := google.CredentialsFromJSON(ctx, []byte(json),
+		"https://www.googleapis.com/auth/cloud-platform",
+		"https://www.googleapis.com/auth/compute")
+	if err != nil {
+		return nil, fmt.Errorf("could not obtain OAuth2 token from JSON: %w", err)
+	}
+	return newOidcAuthorizerForAccountsAPI(oidcSource, creds.TokenSource), nil
```

Comment:
So, this one may work for accounts api, but what about workspace itself? Works for both?

Reply:
From my own testing it works for both.